tests/default.nf.test

```diff
@@ -0,0 +1,274 @@
+    script "../main.nf"
+    tag "pipeline"
+
+    test("-profile test") {
+
+        setup {
+            // Data and Databases: that use curl | tar
+            // For loop as the data is all treated the same
+
+            def database_dict = [
+                                    "TEST_DATA" : [
+                                        "URL": "https://tolit.cog.sanger.ac.uk/test-data/resources/ascc/asccTinyTest_V2.tar.gz",
+                                        "OUT": "asccTinyTest_V2"
+                                    ],
+                                    "BLASTN" : [
+                                        "URL": "https://dp24.cog.sanger.ac.uk/blastn.tar.gz",
+                                        "OUT": "blastn"
+                                    ],
+                                    "BUSCO_LINEAGES" :  [
+                                        "URL": "https://tolit.cog.sanger.ac.uk/test-data/resources/busco/blobtoolkit.GCA_922984935.2.2023-08-03.lineages.tar.gz",
+                                        "OUT": "busco_database"
+                                    ],
+                                    "KRAKEN2" :         [
+                                        "URL": "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/db/kraken2.tar.gz",
+                                        "OUT": "kraken2"
+                                    ],
+                                    "NT_SUBSET" :       [
+                                        "URL": "https://ftp.ncbi.nlm.nih.gov/blast/db/18S_fungal_sequences.tar.gz",
+                                        "OUT": "NT_database"
+                                    ],
+                                    "VECSCREEN" :       [
+                                        "URL": "https://ftp.ncbi.nlm.nih.gov/blast/db/v4/16SMicrobial_v4.tar.gz",
+                                        "OUT": "vecscreen"
+                                    ]
+                                ]
+
+
+            for (String dbKey : database_dict.keySet()) {
+                def db = database_dict.get(dbKey)
+                new File("${projectDir}/${dbKey}").mkdir()
+
+                println "\nDownloading the database ${dbKey}..."
+                def command = ['bash', '-c', "curl ${db['URL']} | tar xzf - -C ${projectDir}/${dbKey}/"]
+                def process = command.execute()
+                process.waitFor()
+
+                if (process.exitValue() != 0) {
+                    throw new RuntimeException("Error - failed to download ${dbKey}: ${process.err.text}")
+                }
+
+                println "Done"
+            }
+
+            // Copy the test data so that we have a fake haplo
+            def data_copy_command = ['bash', '-c', "cp ${projectDir}/TEST_DATA/asccTinyTest_V2/assembly/pyoelii_tiny_testfile_with_adapters.fa ${projectDir}/TEST_DATA/asccTinyTest_V2/assembly/Pyoeliiyoelii17XNL_assembly_hap.fa"]
```

Comment:
And again - download to `launchDir`

Reply:
Updating test data so it gets pulled straight from s3.